aten/src/ATen/native/cpu/PaddingKernel.cpp

```diff
@@ -424,6 +424,8 @@ void replication_pad3d_backward_kernel_impl(
 
 } // anonymous namespace
 
+// These kernels are slower with AVX512 than with AVX2.
+#ifndef CPU_CAPABILITY_AVX512
 // reflection padding
 REGISTER_DISPATCH(reflection_pad1d_kernel, &reflection_pad1d_kernel_impl);
```

Comment:
is it possible to add an default argument to let decide whether this kernel is dispatched to avx512 explicitly ? Probably something like:
```C
// if you want this one to go avx512：
REGISTER_DISPATCH(reflection_pad1d_kernel, &reflection_pad1d_kernel_impl, /* enable_avx512_dispatch */ true);

//  if you want to skip avx512 (use avx2 only), just use the old way：
REGISTER_DISPATCH(reflection_pad1d_kernel, &reflection_pad1d_kernel_impl);
```

Reply:
Agree with @mingfeima and I think it is a better idea to explicitly turn on AVX512 for particular kernels instead of turning off others since there are more kernels to turn off than those to turn on. This can reduces the lines of code changes.

Reply:
Hi, I had started with such an approach when I first started this task, so that the number of changes in the code could be reduced. It leads to symbol resolution issues.

Reply:
Is it possible to always register null AVX512 kernels by default while checking if there are already existing registrations (e.g., via some macro def checks)? The real AVX512 kernels can be registered beforehand.

Reply:
Thanks again for your inputs, @jgong5! Yes, such an approach is feasible.

The approach below to improve readability, and decrease the number of changes is also similar in nature to @mingfeima 's suggestion above.

```
// if you want this one to go avx512：
ALSO_REGISTER_AVX512_DISPATCH(reflection_pad1d_kernel, &reflection_pad1d_kernel_impl, /* enable_avx512_dispatch */ true);

//  if you want to skip avx512 (use avx2 only), just use the old way：
REGISTER_DISPATCH(reflection_pad1d_kernel, &reflection_pad1d_kernel_impl);
```